src/components/ProvidersList.js

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import LocationOnIcon from '@mui/icons-material/LocationOn';
+import {
+  Fab,
+  Divider,
+  Container,
+  Typography,
+  Box,
+} from '@mui/material';
+import { ProdiverPartial } from './ProviderPartial';
+
+export const ProviderList = ({ providers }) => {
+  const [filter, setFilter] = useState('Canada');
+
+  const filterProviders = () => {
+    if (filter === 'Canada') return providers;
+
+    return providers.filter((provider) =>
+      provider.location.includes(filter)
+    );
+  };
+
+  const handleFilterClick = (filterVal) => {
+    setFilter(filter === filterVal ? 'Canada' : filterVal);
+  };
+
+  return (
+    <>
```

Comment:
Would be great if you extracted the loading spinner to the provider list as well, since the loading state here currently is just "zero providers".